### PR TITLE
Add code in javascript and typescript.

### PR DESCRIPTION
**Pull request by gaofer**
Add binary_search_insertion and binary_search_edge code in JavaScript and TypeScript.

If this PR is related to coding or code translation, please fill out the checklist and paste the console outputs to the PR.

There is my console outputs:
- JavaScript
  - binary_search_insertion.js
image
  - binary_search_edge.js
image

- TypeScript
  - binary_search_insertion.ts
image
  - binary_search_edge.ts
image

- [x] I've tested the code and ensured the outputs are the same as the outputs of reference codes.
- [x] I've checked the codes (formatting, comments, indentation, file header, etc) carefully.
- [x] The code does not rely on a particular environment or IDE and can be executed on a standard system (Win, macOS, Ubuntu).

### CODE REVIEW

**Comment by gaofer** [comment]
@justin-tse @krahets Please review my code and feel free to contact me if there are any issues. I will make timely modifications.
请检查我的代码，如果有问题请随时联系我，我将及时作出修改。

**Review by justin-tse — COMMENTED** [review]
Hi @gaofer Thank you for your clean code! Please follow the review and update your code.

**Comment by gaofer** [comment]
Hi @justin-tse Thank you very much for taking time out of your daily life to review my code, which means a lot to me! I have updated my code and will pay attention to these issues in the next submission. Thank you again!
非常感谢你在日常生活中抽出时间来审查我的代码，这对我来说意义重大！我已经更新了我的代码，并将在下次提交时关注这些问题。再次感谢！

File: codes/javascript/chapter_searching/binary_search_edge.js [hunk 1]
```diff
@@ -0,0 +1,45 @@
+/**
+ * File: binary_search_edge.js
+ * Created Time: 2023-08-22
+ * Author: Gaofer Chou (gaofer-chou@qq.com)
+ */
+
+import binarySearchInsertion from './binary_search_insertion.js';
```
Comment (justin-tse) [inline codes/javascript/chapter_searching/binary_search_edge.js:7]:
```suggestion
const { binarySearchInsertion } = require('./binary_search_insertion.js');
```

Comment (justin-tse) [reply codes/javascript/chapter_searching/binary_search_edge.js:7]: Our JS code is using the CommonJS module, so it needs to change.

File: codes/javascript/chapter_searching/binary_search_insertion.js [hunk 2]
```diff
@@ -0,0 +1,60 @@
...
+}
+
+// 包含重复元素的数组
+nums = [1, 3, 6, 6, 6, 6, 6, 10, 12, 15];
+console.log('\n数组 nums = ' + nums);
+// 二分查找插入点
+for (const target of [2, 6, 20]) {
+    const index = binarySearchInsertion(nums, target);
+    console.log('元素 ' + target + ' 的插入点的索引为 ' + index);
+}
```
Comment (justin-tse) [inline codes/javascript/chapter_searching/binary_search_insertion.js:60]:
```suggestion
}


module.exports = {
    binarySearchInsertion
};

```

Comment (justin-tse) [reply codes/javascript/chapter_searching/binary_search_insertion.js:60]: The function needs to export, so we can use it.

**Review by justin-tse — COMMENTED** [review]
@gaofer Thank you for your contribution, please update.

**Review by justin-tse — COMMENTED** [review]
@gaofer Thank you for your contribution!

**Comment by gaofer** [comment]
The newly submitted code has modified unnecessary type definitions and corrected the issue with method export. Please review again. Thank you very much for your patience!

**Review by justin-tse — APPROVED** [review]
@gaofer Good job, thank you for your clean code

**Review by krahets — APPROVED** [review]

**Pull request merged**